Hunting Queries/SecurityEvent/Least_Common_Process_Command_Lines.txt

```diff
@@ -0,0 +1,18 @@
+// Name: Least Common Processes Including Folder Depth
+// Id: 088d30e9-c02b-46b1-bd1f-d5b6d6b782f0
+// Description: Looks across your environment for least common Process Command Lines, may be noisy and require whitelisting.  By ZanCo
+// DataSource: #SecurityEvent
+// Tactics: #InitialAccess, #Execution, #Persistance
+let Whitelist = dynamic (['foo.exe', 'baz.exe']);
+let Sensitivity = 5;
+let StartDate = ago(7d);
+let Duration = 7d;
+SecurityEvent
+| where EventID == 4688 and TimeGenerated > StartDate and TimeGenerated < (StartDate + Duration)
+| extend ProcArray = split(NewProcessName, '\\')
+| extend ProcArrayLength = arraylength(ProcArray)
+| extend LastIndex = ProcArrayLength - 1
+| extend Proc = ProcArray[LastIndex]
+| where Proc !in (Whitelist)
+| summarize TimesSeen = count(), Hosts = makeset(Computer), Processes = makeset(Proc), Users = makeset(SubjectUserName) by CommandLine
+| where TimesSeen < Sensitivity
```

Comment:
Thinking on this one further and it may make sense to source a list of machines that are controlled and should not have random scripts and such running on them.  This query returns thousands of results even when I exclude things like cmd.exe, csc.exe, cscript.exe, cvtres.exe, mpcmdrun.exe, powershell.exe.  To many processes have unique command lines so depending on the type of environment, this is very noisy, even for a hunting query